### PR TITLE
Add variant id to simulate dataset

### PR DESCRIPTION
**Pull request by benjeffery**
Came across this while trying to save a VCF of a simulated callset - seems like it should be in the simulated set?

### CODE REVIEW

**Comment by tomwhite** [comment]
This problem should be fixed by #1021.

**Comment by benjeffery** [comment]
Should we also add `variant_quality` etc?

**Comment by tomwhite** [comment]
Variant IDs and filters are not required variables, which is why the simulated dataset code doesn't set them. We could add them, but perhaps make them optional? Quality values aren't in bgen or plink datasets, only VCF, so making quality optional in the simulated dataset would make sense.

**Comment by benjeffery** [comment]
Yeah, a set of optional additions to the simulation seems to make sense, will change this PR to that.

**Comment by benjeffery** [comment]
Hows this? It allows arbitrary additional fields.

File: sgkit/tests/test_testing.py [hunk 1]
```diff
@@ -29,3 +29,26 @@ def test_simulate_genotype_call_dataset__phased(tmp_path):
...
+        phased=True,
+        additional_variant_fields={
+            "variant_id": np.str,
+            "variant_filter": np.bool,
+            "variant_quality": np.int8,
+            "variant_yummyness": np.float32,
+        },
+    )
+    assert "variant_id" in ds
+    print(ds["variant_id"].dtype)
```
Comment (tomwhite) [inline sgkit/tests/test_testing.py:47]: This print should be removed

**Review by tomwhite — APPROVED** [review]
LGTM. I think the error case needs testing to meet the coverage target.

File: sgkit/tests/test_testing.py (hunk 1 above)
Comment (benjeffery) [reply sgkit/tests/test_testing.py]: Whoops, fixed!

**Comment by benjeffery** [comment]
All fixed up and good to go.

**Comment by benjeffery** [comment]
@tomwhite I have a `vcf_to_zarr` failure here which isn't related to my changes I think.

File: sgkit/testing.py [hunk 2]
```diff
@@ -96,3 +101,18 @@ def simulate_genotype_call_dataset(
...
+            if field_dtype in (np.float32, np.float64):
+                field = rs.rand(n_variant).astype(field_dtype)
+            elif field_dtype in (np.int8, np.int16, np.int32, np.int64):
+                field = rs.randint(0, 100, n_variant, dtype=field_dtype)
+            elif field_dtype is np.bool:
+                field = rs.rand(n_variant) > 0.5
+            elif field_dtype is np.str:
+                field = np.arange(n_variant).astype("S")
+            else:
+                raise ValueError(f"Unrecognized dtype {field_dtype}")
```
Comment (tomwhite) [inline sgkit/testing.py:116]: This line needs a test to meet the 100% coverage requirement

**Review by tomwhite — APPROVED** [review]

**Pull request merged**